### DISCUSSION TITLE
Unexpected data type when conditionally updating column

### DISCUSSION THREAD

**Discussion by frankvollebregt** [simolus3/drift#1790] [Q&A] [answered]
I've written a function in my `.drift` file to update a certain table. One of the columns that is updated is the `status`, which should only ever become larger (and can be ignored if a lower status is pushed). I still want to update the other column values though, so `UPDATE ... WHERE` would not work in my case. I came across [a StackOverflow post](https://stackoverflow.com/a/6401751/15469537) which tries to solve a similar issue. However, when I implement it like this
```sql
updateStatus: UPDATE my_table SET status = MAX(status, :newStatus), second_col = :secondCol;
```
and generate my function, it thinks `newStatus` should be a `String`, while `status` is defined like this:
```sql
CREATE TABLE my_table (
  ...
  status INT NULL
);
```
IDE error message: `The argument type 'int' can't be assigned to the parameter type 'String'`.

And indeed when I look in the generated code, it has set the type to String:
```dart
variables: [
  Variable<String>(newStatus),
  ...
]
```

**Comment by North101** [comment, marked as answer]
You need to give `:newStatus` a type as it seems its not inferred. 

```sql
updateStatus(:newStatus AS INT):
UPDATE my_table SET status = MAX(status, :newStatus), second_col = :secondCol;
```

https://drift.simonbinder.eu/docs/using-sql/drift_files/#variables

**Reply by frankvollebregt** [reply]
Ah, I did not know I could mark them as a certain type. Thank you!

**Reply by simolus3** [reply]
I've also fixed the inaccurate analysis in 645489a21c49a15a859e83362b091558dd1e5f03. If you notice more cases like this, please let me know :)

**Reply by davidmartos96** [reply]
@simolus3 Related to this, I found the following warning a while back.
When using the expression `strftime('%s','now') * 1000`, it warns about strftime not being numeric. `%s` returns the number of seconds since epoch. https://www.sqlite.org/lang_datefunc.html
